packages/connectivity/src/scp-cf/destination/http-proxy-util.ts

```diff
@@ -42,28 +46,30 @@ export function proxyStrategy(destination: HttpDestination): ProxyStrategy {
     return 'private-link';
   }
 
-  const destinationProtocol = getProtocolOrDefault(destination);
-  if (!getProxyEnvValue(destinationProtocol)) {
-    logger.debug(
-      `Could not find proxy settings for ${destinationProtocol} in the environment variables - no proxy used.`
-    );
-    return 'no-proxy';
-  }
+  if (isHttpDestination(destination)) {
```

Comment:
[q] Is this only relevant to HTTP destinations? If I understand the code correctly this is about reading the env variables, shouldn't those apply to the MAIL destinations too? (at least partially, except for the part with the protocol)

Reply:
@marikaner  `getProtocolOrDefault` expects an `HttpDestination` and extracts the protocol from the `url` property (not relevant for the `MAIL` destinations). Based on this protocol, the proxy strategy is decided and if it's `Internet`, the env variables are read and the proxyConfiguration object is built for the destination. However, looking at the mail-client code, we only consider `proxyConfiguration` for `OnPremise` proxy type. So for `Internet`, I don't think mail has to be considered. @jjtang1985 please correct me if my understanding is wrong.

Also, in this [PR](https://github.com/SAP/cloud-sdk-js/pull/2883), we exposed the Transport proxy option so users could decide themselves if they want to pass this.

Reply:
@deekshas8 
Here is the [ADR](https://github.com/SAP/cloud-sdk-js/blob/main/knowledge-base/adr/0033-send-email.md#protocol-of-the-proxy) about the proxy, defined in the env.
Basically, the same logic as the http-client, which is more or less the same as the old implementation.
The changes in this file now specifically changed this.